src/platforms/macos/daemon/dnsutilsmacos.cpp

```diff
@@ -131,14 +123,28 @@ bool DnsUtilsMacos::updateResolvers(const QString& ifname,
   cfDictSetStringList(dnsConfig, kSCPropNetDNSServerAddresses, list);
   cfDictSetString(dnsConfig, kSCPropNetDNSDomainName, "lan");
 
-  // Backup each network service's DNS config, and replace it with ours.
+  // Take a snapshot of the DNS services, and start a process to restore it.
```

Comment:
Can this handle the multi-hop scenario where updateResolvers is called multiple times? It looks like multiple watchdogs will be forked in that case, is that correct?

Reply:
In the multi-hop scenario, `updateResolvers()` is only called when setting up the exit hop, so it should only ever be called once when trying to activate the tunnel.

Reply:
The check for this is done in `Daemon::maybeUpdateResolvers()`